shared-tsconfig.json

```diff
@@ -0,0 +1,11 @@
+{
```

Comment:
I welcome feedback on this shared config. I think most folks should be able to simply do:

```json
{
  "extends": "./node_modules/kcd-scripts/shared-tsconfig.json",
  "include": ["src/*"]
}
```

But I may be mistaken.

Reply:
I'd suggest adding `"forceConsistentCasingInFileNames": true` which
> Disallow inconsistently-cased references to the same file.

And `assumeChangesOnlyAffectDirectDependencies: true` which
> When this option is enabled, TypeScript will avoid rechecking/rebuilding all truly possibly-affected files, and only recheck/rebuild files that have changed as well as files that directly import them.
>
> This can be considered a ‘fast & loose’ implementation of the watching algorithm, which can drastically reduce incremental rebuild times at the expense of having to run the full build occasionally to get all compiler error messages.

Reply:
I think the `include` can even be left out in project-specific `ts-config` if we include it in `shared-tsconfig`, no? 🤔

People can then override `include` if they want to in their project-specific `ts-config` if they want.

Reply:
Thanks @eddyw,

> at the expense of having to run the full build occasionally to get all compiler error messages.

Seems like it would be confusing if we enable this and someone runs typecheck once and gets some error messages, and then they run it again and they don't 🤔

I'm not sure I understand `forceConsistentCasingInFileNames`.

Thank you @MichaelDeBoey,

I was worried that `include` would be relative to the shared config, not the extending config. I'll do a little test to be sure.

Reply:
Yup, my suspicions were confirmed:

![image](https://user-images.githubusercontent.com/1500684/100360798-a8f34280-2fb6-11eb-89e2-08dc504e80c5.png)

I suppose I could try `../../src/*`... That actually should work fine... I think I'll do that.

Reply:
I'm going to enable `forceConsistentCasingInFileNames` but not `assumeChangesOnlyAffectDirectDependencies`. We can alter that in the future if I learn to trust it.

Reply:
> I'm not sure I understand `forceConsistentCasingInFileNames`.

It's like the most valuable option in tsconfig 😆 (joking)
It warns the developer if there is inconsistent casing in the filenames when `import`ing or `require`

For instance, if you have `thisExample.ts` but you do `import {} from './ThisExample.ts`, it'll warn because the casing of the filenames is inconsistent. While this code will likely run in Windows, it'll surely break in Linux / MacOS.

> I was worried that `include` would be relative to the shared config, not the extending config. I'll do a little test to be sure.

The default value for `include` is `**/*` if `files` option isn't specified. Maybe it's a sane default and you could leave it as is?

Also, I'd suggest removing `exclude` since the default value is actually better:
> Default: ["node_modules", "bower_components", "jspm_packages"], plus the value of `outDir` if one is specified.

Reply:
Thanks for that! I only want to `include` the `src` files because those are the ones that I want to generate type defs for. People can override if they need.

I'm pushing a fix to remove `exclude`. Thanks!